vmm/src/config.rs

```diff
@@ -1603,6 +1613,95 @@ impl FsConfig {
     }
 }
 
+#[cfg(feature = "fw_cfg")]
+impl FwCfgConfig {
+    pub const SYNTAX: &'static str = "Boot params to pass to FW CFG device \
+    \"disable<optional; enabled if not set>,e820=on|off,kernel=on|off,cmdline=on|off, \
+    initramfs=on|off,acpi_table=on|off,items=[name0=<backing_file_path>,file0=<file_path>:name1=<backing_file_path>,file1=<file_path>]\"";
```

Comment:
Are there reasons where 

- one provides `--kernel` and `--cmdline`
- fw_cfg device is used, and
- this information IS NOT passed to fw_cfg?

I'm concerned we might are going fully feature-creep here? If I'm not missing something, I'd like to make this dependend on whether `--kernel` et. al. are avaialble

Reply:
Yes, that's a potential senario as intended. It all depends on what kind of use cases we want to offer to users with `fw_cfg`.

You are right, all booting information needs to be passed to `fw_cfg`, when it is used for booting with oak firmware (Google's main use case for confidential computing) and potentially with native OVMF (we require changes to edk2 to support firmware boot with CH now). 

If you look at how the `fw_cfg` is being used and tested (e.g. the integration test). It is merely for passing a file to the guest, which practically is the only use case we support with this PR.

Until we support and decide to narrow down the `fw_cfg` use case to firmware boot. The current CLI options make more sense to me.